packages/ui/core-components/src/lib/atoms/inputs/dropdown/Dropdown.svelte

```diff
@@ -39,162 +38,199 @@
+	export let value,
+		data,
+		label,
+		order = undefined,
+		where = undefined;
 	/** @type {import("@evidence-dev/component-utilities/buildQuery.js").QueryProps}*/
 	const { results, update } = buildReactiveInputQuery(
 		{ value, data, label, order, where },
 		`Dropdown-${name}`,
-		$page.data.data[`Dropdown-${name}`]
+		$page?.data?.data[`Dropdown-${name}`]
 	);
 	$: update({ value, data, label, order, where });
+	/** @type {{ hasQuery: boolean, query: import("@evidence-dev/sdk/usql").QueryValue }}*/
 	$: ({ hasQuery, query } = $results);
 
-	const exec = getQueryFunction();
-	function updateItems(search) {
-		items =
-			search && hasQuery
-				? Query.create(
-						`
-						SELECT
-							*,
-							jaro_winkler_similarity(lower('${search.replaceAll("'", "''")}'), lower(label)) as similarity
-						FROM (${query.text}) WHERE similarity > 0.5 ORDER BY similarity DESC`,
-						exec,
-						{
-							initialData: $items ?? $query,
-							initialDataDirty: true,
-							id: `Dropdown-${name}-searched-${search}`
-						}
-					)
-				: $query;
-	}
-
-	function useNewQuery(query) {
-		items = query;
+	/** @type {import("@evidence-dev/sdk/usql").QueryValue} */
+	let queryOptions;
 
-		if (hasQuery && hasDefault) {
-			ctx.hasBeenSet = true;
-			if (browser) {
-				(async () => {
-					await query.fetch();
-					$selectedValues = query.filter((x) => _defaultValue.find((d) => x.value == d));
-					selectedValuesToInput();
-				})();
-			} else {
-				$selectedValues = query.filter((x) => _defaultValue.find((d) => x.value == d));
-				selectedValuesToInput();
-			}
+	// Uses context under the hood
+	const updateQueryOptions = debounce(async () => {
+		if (search && hasQuery) {
+			// TODO: This may fall victim to the race condition
+			// We need a canonical way / function that handles query replacement like this (e.g. updating buildReactiveInputQuery)
+			const searchQ = query.search(search, 'label');
+			await searchQ.fetch();
+			queryOptions = searchQ;
 		} else {
-			ctx.hasBeenSet = false;
+			queryOptions = query;
 		}
-	}
+	}, 250);
 
-	let items;
-	$: useNewQuery($query);
+	// Update the items when the query changes
+	$: search, query, updateQueryOptions();
 
-	const debouncedUpdateItems = debounce(updateItems, 250);
-	$: debouncedUpdateItems(search);
-</script>
+	/**
+	 * Resets the defaults whenever parameters change
+	 */
+	function evalDefaults() {
+		resolveMaybePromise(
+			() => {
+				if ($selectedOptions.length) {
+					// Try to keep any current selection before falling back to defaults
+					// e.g. if the passed-in query has changed, but the user had a previous selection
+					const presentValues = $selectedOptions.filter((x) =>
+						$options.some((o) => o.value === x.value && o.label === x.label)
+					);
+					// If the available values have changed (e.g. one is missing in the new query), remove that one and update the selection
+					if (presentValues.length !== $selectedOptions.length) $selectedOptions = presentValues;
```

Comment:
I don't know if length is a good heuristic to detect change, they could be using the top 10 of something